packages/manager/apps/pci-databases-analytics/src/pages/services/[serviceId]/users/_components/ToggleAcl.component.tsx

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Label } from '@/components/ui/label';
+import { Switch } from '@/components/ui/switch';
+import { useServiceData } from '../../Service.context';
+import { useToast } from '@/components/ui/use-toast';
+import { useEditService } from '@/hooks/api/database/service/useEditService.hook';
+import * as database from '@/types/cloud/project/database';
+
+const ToggleAcl = () => {
+  const { t } = useTranslation(
+    'pci-databases-analytics/services/service/users',
+  );
+  const { projectId, service, serviceQuery } = useServiceData();
+  const aclsEnabled = !!('aclsEnabled' in service && service.aclsEnabled);
+  const [switchState, setSwitchState] = useState(aclsEnabled);
+
+  const toast = useToast();
+  const { editService, isPending } = useEditService({
+    onError: (err) => {
+      toast.toast({
+        title: t('toggleAclErrorTitle'),
+        variant: 'destructive',
+        description: err.response.data.message,
+      });
+      setSwitchState(aclsEnabled);
+    },
+    onEditSuccess: (updatedService) => {
+      toast.toast({
+        title: t('toggleAclSuccessTitle'),
+        description: ((updatedService as unknown) as database.opensearch.Service)
```

Comment:
Why to cast the type coming from the `onError` method from the `UseEditService` hook?
Could be error prone